src/sidebar/components/filter-status.js

```diff
@@ -0,0 +1,153 @@
+import { createElement } from 'preact';
+
+import { countVisible } from '../util/thread';
+
+import Button from './Button';
+
+import useRootThread from './hooks/use-root-thread';
+import useStore from '../store/use-store';
+
+/**
+ * Render information about the currently-applied filters and allow the
+ * user to clear filters or toggle user-focus mode on and off
+ *
+ * When any filtering is applied, we are in one of 3 mutually-exclusive filtering
+ * "modes". In any mode, we display descriptive text about what the user
+ * is looking at, and present one button to take action on the current filtering
+ * state:
+ * - selection: One or more annotations have been selected in the document by
+ *   the user, or we're looking at a direct-linked annotation. Button clears
+ *   the selection. This mode supersedes all others (i.e. other filters ignored).
+ * - query: There is a user-entered filter query. It can "stack" with user-focus.
+ *    Button clears the selection (query, etc.), but will leave user-focus mode
+ *    intact.
+ * - focus: Neither of the above two modes apply and user-focus mode is
+ *   configured (it may or may not be active). Button toggles activation of
+ *   user-focus mode.
+ *
+ * Descriptive text about the filter status and annotations follows the
+ * pattern:
+ *
+ *   [Showing] (<resultCount>|No) (annotation[s]|result[s]) [for <filterQuery>]
+ *   [by <focusedUser>] [\(and <forcedCount> more\)]`
+ */
+export default function FilterStatus() {
+  const rootThread = useRootThread();
+
+  const filterState = useStore(store => store.filterState());
+
+  // The total count of all of the annotations in the store—may differ from
+  // the number of annotations in the thread and the number of visible annotations
+  const totalCount = useStore(store => store.annotationCount());
+
+  // Actions
+  const clearSelection = useStore(store => store.clearSelection);
+  const toggleFocusMode = useStore(store => store.toggleFocusMode);
+
+  const filterMode = (state => {
+    if (state.selectedCount > 0) {
+      return 'selection';
+    } else if (state.filterQuery) {
+      return 'query';
+    } else if (state.focusConfigured) {
+      return 'focus';
+    }
+    return null;
+  })(filterState);
```

Comment:
I can't see the need to pass in `filterState` here, you can just reference it directly in the closure, or put `const state = filterState` at the top if you need a shorthand.

Reply:
I was just trying to encapsulate the scope, but I suppose it wasn't necessary.